pandas/tseries/timedeltas.py

```diff
@@ -142,13 +137,50 @@ def _validate_timedelta_unit(arg):
 
 
 def _coerce_scalar_to_timedelta_type(r, unit='ns', box=True, errors='raise'):
-    """
-    convert strings to timedelta; coerce to Timedelta (if box), else
-    np.timedelta64
-    """
+    """Convert string 'r' to a timedelta object."""
+
+    try:
+        result = tslib.convert_to_timedelta64(r, unit)
+    except ValueError:
+        if errors == 'raise':
+            raise
+        elif errors == 'ignore':
+            return r
+
+        # coerce
+        result = pd.NaT
 
-    result = tslib.convert_to_timedelta(r, unit, errors)
     if box:
         result = tslib.Timedelta(result)
-
     return result
+
+
+def _convert_listlike(arg, unit='ns', box=True, errors='raise', name=None):
+    """Convert a list of objects to a timedelta index object."""
+
+    if isinstance(arg, (list, tuple)) or not hasattr(arg, 'dtype'):
+        arg = np.array(list(arg), dtype='O')
+
+    # these are shortcut-able
+    if is_timedelta64_dtype(arg):
+        value = arg.astype('timedelta64[ns]')
+    elif is_integer_dtype(arg):
+        value = arg.astype('timedelta64[{0}]'.format(
+            unit)).astype('timedelta64[ns]', copy=False)
+    else:
+        try:
+            value = tslib.array_to_timedelta64(_ensure_object(arg),
+                                               unit=unit, errors=errors)
+            value = value.astype('timedelta64[ns]', copy=False)
+        except ValueError:
+            if errors == 'ignore':
+                return arg
+            else:
+                # all errors should be raised ('raise') OR
```

Comment:
you don't need the else.

I don't understand your comment (well the first sentence makes sense), but the 2nd doesn't. if we are coercing how could we ever get here?